File: src/cpp/Source/Chunks/Shex/InterpolationMode.cpp
#include "PCH.h"
#include "InterpolationMode.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(InterpolationMode value)
{
	switch (value)
	{
	case InterpolationMode::Constant :
		return "constant";
	case InterpolationMode::Linear :
		return "linear";
	case InterpolationMode::LinearCentroid :
		return "linear centroid";
	case InterpolationMode::LinearNoPerspective :
		return "linear noperspective";
	case InterpolationMode::LinearNoPerspectiveCentroid :
		return "linear noperspective centroid";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}

File: src/csharp/SlimShader/Chunks/Shex/InterpolationMode.cs
namespace SlimShader.Chunks.Shex
{
	public enum InterpolationMode
	{
		Undefined = 0,

		[Description("constant")]
		Constant = 1,

		[Description("linear")]
		Linear = 2,

		[Description("linear centroid")]
		LinearCentroid = 3,

		[Description("linear noperspective")]
		LinearNoPerspective = 4,

		[Description("linear noperspective centroid")]
		LinearNoPerspectiveCentroid = 5,

		// Following are new in DX10.1

		LinearSample = 6,
		LinearNoPerspectiveSample = 7
	}
}